parser/src/parser/tests/list_comprehension.rs

```diff
@@ -440,455 +237,193 @@ fn ic_one_iterable_identifier_lc() {
-                    SymbolAccess(SymbolAccess::new(
-                        Identifier("c".to_string()),
-                        AccessType::Default,
-                        0,
-                    )),
-                    vec![(
-                        Identifier("c".to_string()),
-                        Iterable::Slice(Identifier("c".to_string()), Range::new(0, 3)),
-                    )],
-                )),
-            )),
-            IntegrityStmt::Constraint(IntegrityConstraint::new(
-                ConstraintExpr::Inline(InlineConstraintExpr::new(
-                    SymbolAccess(SymbolAccess::new(
-                        Identifier("a".to_string()),
-                        AccessType::Default,
-                        0,
-                    )),
-                    Add(
-                        Box::new(Add(
-                            Box::new(Add(
-                                Box::new(SymbolAccess(SymbolAccess::new(
-                                    Identifier("x".to_string()),
-                                    AccessType::Vector(0),
-                                    0,
-                                ))),
-                                Box::new(SymbolAccess(SymbolAccess::new(
-                                    Identifier("x".to_string()),
-                                    AccessType::Vector(1),
-                                    0,
-                                ))),
-                            )),
-                            Box::new(SymbolAccess(SymbolAccess::new(
-                                Identifier("x".to_string()),
-                                AccessType::Vector(2),
-                                0,
-                            ))),
-                        )),
-                        Box::new(SymbolAccess(SymbolAccess::new(
-                            Identifier("x".to_string()),
-                            AccessType::Vector(3),
-                            0,
-                        ))),
-                    ),
-                )),
-                None,
-                None,
-            )),
-        ]),
-    ]);
-
-    ParseTest::new().expect_ast(source, expected);
+    let mut expected = Module::new(ModuleType::Root, SourceSpan::UNKNOWN, ident!(test));
+    expected
+        .trace_columns
+        .push(trace_segment!(0, "$main", [(a, 1), (b, 1), (c, 4)]));
+    expected.public_inputs.insert(
+        ident!(inputs),
+        PublicInput::new(SourceSpan::UNKNOWN, ident!(inputs), 2),
+    );
+    expected.boundary_constraints = Some(Span::new(
+        SourceSpan::UNKNOWN,
+        vec![enforce!(eq!(bounded_access!(a, Boundary::First), int!(0)))],
+    ));
+    expected.integrity_constraints = Some(Span::new(
+        SourceSpan::UNKNOWN,
+        vec![
+            let_!(x = lc!(((c, expr!(slice!(c, 0..3)))) => access!(c)).into() =>
+                   enforce!(eq!(access!(a), add!(add!(add!(access!(x[0]), access!(x[1])), access!(x[2])), access!(x[3]))))),
```

Comment:
note: this shouldn't be valid, because the range syntax should not be inclusive, so x only has length 3